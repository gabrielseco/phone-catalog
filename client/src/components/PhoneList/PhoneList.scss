@import '~styles/settings/global';
@import '~styles/tools/mixins/all';

.container {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  @include respond-to(medium) {
    flex-direction: row;
  }
}
