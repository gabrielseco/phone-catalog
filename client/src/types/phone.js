// @flow
export type Phone = {
  id: number,
  name: string,
  manufacturer: string,
  price: string,
  image: string
};
