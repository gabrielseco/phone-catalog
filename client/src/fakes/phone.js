// @flow
import { type Phone } from './../types/phone';

export const phonesFake: Phone[] = [
  {
    id: 1,
    name: 'Iphone X',
    manufacturer: 'Apple Inc',
    price: '999$',
    image: 'https://img.shoppydoo.es/products/7/apple_iphone_x_64gb-300x300.jpg'
  },
  {
    id: 2,
    name: 'Samsung Galaxy S9',
    manufacturer: 'Samsung',
    price: '899$',
    image:
      'https://www.satugadget.com.my/content/images/thumbs/0008898_samsung-galaxy-s9-plus-256gb-purple-original-malaysia-set_300.jpeg'
  }
];
