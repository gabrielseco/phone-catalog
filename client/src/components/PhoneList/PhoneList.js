// @flow
import React from 'react';
import { PhoneListItem } from 'components';
import { type Phone } from './../../types/phone';
import styles from './PhoneList.scss';

type PhoneListProps = {
  phones: Phone[]
};

const PhoneList = (props: PhoneListProps) => {
  const phones = props.phones.map((phone, index) => {
    return <PhoneListItem phone={phone} key={index} />;
  });

  return (
    <div role="list" className={styles.container}>
      {phones}
    </div>
  );
};

export default PhoneList;
