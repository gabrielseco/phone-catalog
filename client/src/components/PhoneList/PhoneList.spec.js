import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import PhoneList from './PhoneList';
import { phonesFake } from 'fakes/phone';

describe('PhoneList suite', () => {
  it('should render the ProductsList component', () => {
    const jsxComponent = (
      <Router>
        <PhoneList phones={phonesFake} />
      </Router>
    );
    const component = shallow(<PhoneList phones={phonesFake} />);
    const tree = renderer.create(jsxComponent).toJSON();
    expect(component).toBeDefined();
    expect(tree).toMatchSnapshot();
  });
});
