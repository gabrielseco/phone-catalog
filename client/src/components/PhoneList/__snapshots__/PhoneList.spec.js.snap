// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneList suite should render the ProductsList component 1`] = `
<div
  className="container"
  role="list"
>
  <div
    className="container phone-list-item"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="listitem"
    tabIndex={0}
  >
    <img
      alt="Iphone X"
      className="phoneImg"
      id="img-1"
      src="https://img.shoppydoo.es/products/7/apple_iphone_x_64gb-300x300.jpg"
    />
    <h2
      className="title title"
    >
      Iphone X
    </h2>
    <p
      className="price price"
    >
      999$
    </p>
    <a
      className="btnAction"
      href="#"
      id="button-1"
      onClick={[Function]}
    >
      See more
    </a>
  </div>
  <div
    className="container phone-list-item"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="listitem"
    tabIndex={0}
  >
    <img
      alt="Samsung Galaxy S9"
      className="phoneImg"
      id="img-2"
      src="https://www.satugadget.com.my/content/images/thumbs/0008898_samsung-galaxy-s9-plus-256gb-purple-original-malaysia-set_300.jpeg"
    />
    <h2
      className="title title"
    >
      Samsung Galaxy S9
    </h2>
    <p
      className="price price"
    >
      899$
    </p>
    <a
      className="btnAction"
      href="#"
      id="button-2"
      onClick={[Function]}
    >
      See more
    </a>
  </div>
</div>
`;
